lib/Result.test.ts

```diff
@@ -221,4 +221,16 @@ describe("Result tests", () => {
       await currentResult.inverted().flatMapSuccess(() => failure("failed" as const))
     ).toMatchObject({ status: "failure" as const, value: "failed" as const })
   })
+  
+  it("should allow an array of promise results to resolve as one success result containing the array of success values", async () => {
+    expect(
+      await promiseResultAllSettled([promiseResult(success("success1" as const)), promiseResult(success("success2" as const))])
+    ).toMatchObject({ status: "success" as const, value: ["success1", "success2"] })
+  })
+  
+  it("should allow an array of promise results to resolve as one failure result containing the array of failure values", async () => {
+    expect(
+      await promiseResultAllSettled([promiseResult(success("success1" as const)), promiseResult(failure("failed1" as const)), promiseResult(failure("failed2" as const))])
+    ).toMatchObject({ status: "failure" as const, value: ["failed1", "failed2"] as const })
```

Comment:
Shouldn't this return a singular success result that contains all the result values (including success values in this case). This would be more in line with what `Promise.allSettled` does.

So, maybe it could work like:
```ts
const result = await promiseResultAllSettled([
  promiseResult(success("yay")),
  promiseResult(failure("nay"))
])
expect(result).toMatchObject({ 
  status: "success",  
  value: [{ status: "success", value: "yay" }, { status: "failure", value: "nay" }]
})
```

`Promise.allSettled` never throws, but rather it returns each promise with a status of whether or not it failed or resolved. Therefore, with `PromiseResult` we could do a similar thing by always returning a success result with an array of each result as the value. Though admittedly, this also depends a bit on how you want to use this utility, but regardless I don't think it matches well with `Promise.allSettled`.